apps/condo/domains/billing/schema/BillingAccount.test.js

```diff
@@ -510,4 +522,157 @@ describe('BillingAccount', () => {
             }, `${UNEQUAL_CONTEXT_ERROR}:property:context] Context is not equal to property.context`)
         })
     })
-})
\ No newline at end of file
+
+    describe('DiscoverServiceConsumers hook', () => {
+        let context
+        beforeAll(async ()=>{
+            const { context: billingContext }  = await makeContextWithOrganizationAndIntegrationAsAdmin(
+                {}, {}, { status: CONTEXT_FINISHED_STATUS }
+            )
+            context = billingContext
+        })
+        test('should create one ServiceConsumer when BillingAccount is created', async () => {
+            const user = await makeClientWithResidentUser()
+            const unitName = faker.random.alphaNumeric(8)
+            const { address, addressMeta } = buildFakeAddressAndMeta(false)
+            const [organization] = await registerNewOrganization(admin)
+            const [acquiringIntegration] = await createTestAcquiringIntegration(admin)
+            await createTestAcquiringIntegrationContext(admin, organization, acquiringIntegration, { status: CONTEXT_FINISHED_STATUS })
+            const [billingProperty] = await createTestBillingProperty(admin, context, { address })
+            await createTestProperty(admin, organization, {
+                address,
+                addressMeta,
+            })
+            const [resident] = await registerResidentByTestClient(user,
+                { address, addressMeta, unitName, unitType: FLAT_UNIT_TYPE }
+            )
+
+            const [billingAccount] = await createTestBillingAccount(admin, context, billingProperty,
+                { unitName, unitType: FLAT_UNIT_TYPE }
+            )
+
+            await waitFor(async () => {
+                const createdServiceConsumers = await ServiceConsumer.getAll(admin, {
+                    resident: { id: resident.id, deletedAt: null },
+                    accountNumber: billingAccount.number,
+                    deletedAt: null,
+                })
+
+                expect(createdServiceConsumers).toHaveLength(1)
+                expect(createdServiceConsumers[0].organization.id).toEqual(organization.id)
+                expect(createdServiceConsumers.map((serviceConsumer) => serviceConsumer.accountNumber).toString()).toContain(billingAccount.number)
+                expect(createdServiceConsumers[0].accountNumber).toEqual(billingAccount.number)
+            })
+        })
+
+        test('should create multiple ServiceConsumers when BillingAccount is created', async () => {
```

Comment:
Be more specific in cases...Something like 'Must create ServiceConsumer for all existing residents by address and account number'